gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -118,20 +118,41 @@ static void netCDFWriteAttributesFromConf(
         }
         else if( EQUAL(oAtt.m_osType, "string") )
         {
-            status = nc_put_att_text(cdfid, varid, oAtt.m_osName,
-                                     oAtt.m_osValue.size(), oAtt.m_osValue);
+            if(m_bLegacyCreateMode)
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation